### PR TITLE
pixelorama: init at 0.10.1

### PR DESCRIPTION
**Pull request by felschr**
###### Description of changes
Adds Pixelorama: A free & open-source 2D sprite editor, made with the Godot Engine!

Based on @jrobsonchase's [pixelorama derivation](https://github.com/jrobsonchase/nixos-config/blob/9c786901417c6aeee8690f9bf5a903fd2732c811/overlay/pixelorama.nix).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/pixelorama/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  buildPhase = ''
+    export HOME="$(pwd)"
+    mkdir -p .local/share/godot/
+    ln -s "${godot-export-templates}/share/godot/templates" ".local/share/godot/templates"
+    mkdir -p build
+    godot-headless -v --export "Linux/X11 64-bit" ./build/pixelorama
+    godot-headless -v --export-pack "Linux/X11 64-bit" ./build/pixelorama.pck
+  '';
+
+  installPhase = ''
```
Comment (legendofmiracles) [inline pkgs/applications/editors/pixelorama/default.nix:26]: Please run the pre- and postInstall phases

File: pkgs/applications/editors/pixelorama/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{ lib , stdenv , fetchurl , godot-headless , godot-export-templates , writeShellScript }:
+
+stdenv.mkDerivation rec {
+  pname = "pixelorama";
+  version = "0.10.1";
+
+  src = fetchurl {
```
Comment (legendofmiracles) [inline pkgs/applications/editors/pixelorama/default.nix:7]: use fetchFromGitHub instead

File: pkgs/applications/editors/pixelorama/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    godot-headless
+    godot-export-templates
+  ];
+
+  buildPhase = ''
+    export HOME="$(pwd)"
+    mkdir -p .local/share/godot/
+    ln -s "${godot-export-templates}/share/godot/templates" ".local/share/godot/templates"
+    mkdir -p build
+    godot-headless -v --export "Linux/X11 64-bit" ./build/pixelorama
```
Comment (legendofmiracles) [inline pkgs/applications/editors/pixelorama/default.nix:22]: Can it also be built on other platforms? Maybe you can make it dynamic with `stdenv.hostPlatform.linuxArch`

If not meta.platform should be “x86_64-linux” only

File: pkgs/applications/editors/pixelorama/default.nix (hunk 1 above)
Comment (legendofmiracles) [reply pkgs/applications/editors/pixelorama/default.nix:26]: Same for the buildPhase

File: pkgs/applications/editors/pixelorama/default.nix (hunk 3 above)
Comment (felschr) [reply pkgs/applications/editors/pixelorama/default.nix]: I've adapted it to build on other platforms now, but I have no devices to test this. We will need to wait for the ofborg results to see if it works.

Comment (felschr) [reply pkgs/applications/editors/pixelorama/default.nix]: Godot itself is currently just built for `i686-linux` & `x86_64-linux` in nixpkgs, thus I've adjusted the platforms.

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/applications/editors/pixelorama/default.nix [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+  nativeBuildInputs = [
+    godot-headless
+    godot-export-templates
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    export HOME="$(pwd)"
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/pixelorama/default.nix:29]:
```suggestion
    export HOME="$(mktemp -d)"
```

File: pkgs/applications/editors/pixelorama/default.nix [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  src = fetchFromGitHub {
+    owner = "Orama-Interactive";
+    repo = "Pixelorama";
+    rev = "v${version}";
+    sha256 = "sha256-+Sfhv66skHawe6jzfzQyFxejN5TvTdmWunzl0/7yy4M=";
+  };
+
+  nativeBuildInputs = [
+    godot-headless
+    godot-export-templates
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/pixelorama/default.nix:23]:
```suggestion
```
This only adds things to PATH and does not make the copy below cross compile friendlier.

**Pull request merged**